### ISSUE TITLE
Display of grids

### ISSUE THREAD

**Issue by benoitgaudou** [gama-platform/gama#1597]
When implementing and displaying a model with grid agent, there is some issues with the displays:
- in java2D displays, the coloured cells are sometimes a little bit translated (depending on the zoom). And on the borders, some pieces of cells are displayed.
- in opengl, wrong locations of colored cells are even worse.

<img width="309" alt="screen shot 2016-02-29 at 16 28 56" src="https://cloud.githubusercontent.com/assets/1555274/13399050/9c1825fe-df01-11e5-96f3-67e0ab532262.png">

<img width="523" alt="screen shot 2016-02-29 at 16 29 18" src="https://cloud.githubusercontent.com/assets/1555274/13399053/a057d498-df01-11e5-937f-f0ce0b1c3386.png">

The model:

model TestNeigh6

global {
	init {
		ask (cell[5,5]) {
			color <- #yellow;
			rgb c <- rnd_color(255);
			list<cell> lc;
			// lc <- self neighbors_at 2;
			using topology(self){
				lc <- self neighbors_at 2;
			}
			write "" + length(lc) + " -> " + lc;
			ask lc {
				color <- c;
			}				
		}
	}
}

grid cell height: 10 width: 10 neighbors: 4{
	rgb color <- #grey;
}

experiment TestNeigh6 type: gui {
	output {
		display d type: opengl {
			grid cell lines: #black;
		}
	}
}

**Comment by AlexisDrogoul** [comment]
This has been corrected for Java2D (a problem of rounding of double values), but it's a different problem for OpenGL, where the cause is that the width and height of the texture used for drawing the grid are not powers of 2. You can try (and see the difference) if you give your grid a dimension of 16x16. This relates to  issue #1310, which has been closed because there is not a lot we can do on this matter…

**Comment by AlexisDrogoul** [comment]
Turns out I was too pessimistic: I have been able to enable the possibility to load non-power of 2 textures in OpenGL, but this may not be available on some graphic cards. So I have chosen to enable it, and to add a new preference ("Force the dimensions of OpenGL to power of 2"), so that users can toggle it off if nothing gets displayed. I have also been able to detect (not sure it is completely accurate) whether or not the graphic card supports it, which should allow for a finer control.

So, basically, the preference is false by default (i.e. the user does not force anything and lets GAMA decide for him/her), but it can be enabled if there are problems. This should fix both this issue and #1310.
